### DISCUSSION TITLE
Update failed - from v2.8.9 to v2.11.12

### DISCUSSION THREAD

**Discussion by Justinli2030** [netbox-community/netbox#9444] [Q&A] [answered]
Dear All,

When updating from netbox 2.8.9 to 2.11.12, I came across below error after following along the update guide. See below for a detailed output. After reading through a few posts, I might need to add a package to requirements but not sure what I am missing here? Any suggestion would be appreciated.

**ImportError: cannot import name 'NetBoxModelFilterSetForm' from 'netbox.forms' (/opt/netbox-2.11.12/netbox/netbox/forms.py)**

admin@netbox:/opt/netbox$ sudo ./upgrade.sh
Creating a new virtual environment at /opt/netbox-2.11.12/venv...
Updating pip (pip install --upgrade pip)...
Collecting pip
  Using cached pip-22.1.1-py3-none-any.whl (2.1 MB)
Installing collected packages: pip
  Attempting uninstall: pip
    Found existing installation: pip 20.0.2
    Uninstalling pip-20.0.2:
      Successfully uninstalled pip-20.0.2
Successfully installed pip-22.1.1
pip 22.1.1 from /opt/netbox-2.11.12/venv/lib/python3.8/site-packages/pip (python 3.8)
Installing Python system packages (pip install wheel)...
Collecting wheel
  Using cached wheel-0.37.1-py2.py3-none-any.whl (35 kB)
Installing collected packages: wheel
Successfully installed wheel-0.37.1
Installing core dependencies (pip install -r requirements.txt)...
Collecting Django==3.2.6
  Using cached Django-3.2.6-py3-none-any.whl (7.9 MB)
Collecting django-cacheops==6.0
  Using cached django_cacheops-6.0-py2.py3-none-any.whl (39 kB)
Collecting django-cors-headers==3.8.0
  Using cached django_cors_headers-3.8.0-py3-none-any.whl (12 kB)
Collecting django-debug-toolbar==3.2.2
  Using cached django_debug_toolbar-3.2.2-py3-none-any.whl (200 kB)
Collecting django-filter==2.4.0
  Using cached django_filter-2.4.0-py3-none-any.whl (73 kB)
Collecting django-mptt==0.13.1
  Using cached django_mptt-0.13.1-py3-none-any.whl (132 kB)
Collecting django-pglocks==1.0.4
  Using cached django_pglocks-1.0.4-py3-none-any.whl
Collecting django-prometheus==2.1.0
  Using cached django_prometheus-2.1.0-py2.py3-none-any.whl (28 kB)
Collecting django-rq==2.4.1
  Using cached django_rq-2.4.1-py2.py3-none-any.whl (45 kB)
Collecting django-tables2==2.4.0
  Using cached django_tables2-2.4.0-py2.py3-none-any.whl (91 kB)
Collecting django-taggit==1.5.1
  Using cached django_taggit-1.5.1-py3-none-any.whl (53 kB)
Collecting django-timezone-field==4.1.2
  Using cached django_timezone_field-4.1.2-py3-none-any.whl (9.6 kB)
Collecting djangorestframework==3.12.4
  Using cached djangorestframework-3.12.4-py3-none-any.whl (957 kB)
Collecting drf-yasg[validation]==1.20.0
  Using cached drf_yasg-1.20.0-py2.py3-none-any.whl (1.6 MB)
Collecting gunicorn==20.1.0
  Using cached gunicorn-20.1.0-py3-none-any.whl (79 kB)
Collecting Jinja2==3.0.1
  Using cached Jinja2-3.0.1-py3-none-any.whl (133 kB)
Collecting Markdown==3.3.4
  Using cached Markdown-3.3.4-py3-none-any.whl (97 kB)
Collecting netaddr==0.8.0
  Using cached netaddr-0.8.0-py2.py3-none-any.whl (1.9 MB)
Collecting Pillow==8.3.1
  Using cached Pillow-8.3.1-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.whl (3.0 MB)
Collecting psycopg2-binary==2.9.1
  Using cached psycopg2_binary-2.9.1-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.4 MB)
Collecting pycryptodome==3.10.1
  Using cached pycryptodome-3.10.1-cp35-abi3-manylinux2010_x86_64.whl (1.9 MB)
Collecting PyYAML==5.4.1
  Using cached PyYAML-5.4.1-cp38-cp38-manylinux1_x86_64.whl (662 kB)
Collecting svgwrite==1.4.1
  Using cached svgwrite-1.4.1-py3-none-any.whl (66 kB)
Collecting tablib==3.0.0
  Using cached tablib-3.0.0-py3-none-any.whl (47 kB)
Collecting asgiref<4,>=3.3.2
  Using cached asgiref-3.5.2-py3-none-any.whl (22 kB)
Collecting pytz
  Using cached pytz-2022.1-py2.py3-none-any.whl (503 kB)
Collecting sqlparse>=0.2.2
  Using cached sqlparse-0.4.2-py3-none-any.whl (42 kB)
Collecting redis>=3.0.0
  Using cached redis-4.3.1-py3-none-any.whl (241 kB)
Collecting funcy<2.0,>=1.8
  Using cached funcy-1.17-py2.py3-none-any.whl (33 kB)
Collecting six>=1.4.0
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting django-js-asset
  Using cached django_js_asset-2.0.0-py3-none-any.whl (4.9 kB)
Collecting prometheus-client>=0.7
  Using cached prometheus_client-0.14.1-py3-none-any.whl (59 kB)
Collecting rq>=1.2
  Using cached rq-1.10.1-py2.py3-none-any.whl (70 kB)
Collecting ruamel.yaml>=0.15.34
  Using cached ruamel.yaml-0.17.21-py3-none-any.whl (109 kB)
Collecting coreschema>=0.0.4
  Using cached coreschema-0.0.4-py3-none-any.whl
Collecting inflection>=0.3.1
  Using cached inflection-0.5.1-py2.py3-none-any.whl (9.5 kB)
Collecting uritemplate>=3.0.0
  Using cached uritemplate-4.1.1-py2.py3-none-any.whl (10 kB)
Collecting coreapi>=2.3.3
  Using cached coreapi-2.3.3-py2.py3-none-any.whl (25 kB)
Collecting packaging
  Using cached packaging-21.3-py3-none-any.whl (40 kB)
Collecting swagger-spec-validator>=2.1.0
  Using cached swagger_spec_validator-2.7.4-py2.py3-none-any.whl (27 kB)
Requirement already satisfied: setuptools>=3.0 in ./venv/lib/python3.8/site-packages (from gunicorn==20.1.0->-r requirements.txt (line 15)) (44.0.0)
Collecting MarkupSafe>=2.0
  Using cached MarkupSafe-2.1.1-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (25 kB)
Collecting requests
  Using cached requests-2.27.1-py2.py3-none-any.whl (63 kB)
Collecting itypes
  Using cached itypes-1.2.0-py2.py3-none-any.whl (4.8 kB)
Collecting async-timeout>=4.0.2
  Using cached async_timeout-4.0.2-py3-none-any.whl (5.8 kB)
Collecting deprecated>=1.2.3
  Using cached Deprecated-1.2.13-py2.py3-none-any.whl (9.6 kB)
Collecting pyparsing!=3.0.5,>=2.0.2
  Using cached pyparsing-3.0.9-py3-none-any.whl (98 kB)
Collecting click>=5.0.0
  Using cached click-8.1.3-py3-none-any.whl (96 kB)
Collecting ruamel.yaml.clib>=0.2.6
  Using cached ruamel.yaml.clib-0.2.6-cp38-cp38-manylinux1_x86_64.whl (570 kB)
Collecting jsonschema
  Using cached jsonschema-4.5.1-py3-none-any.whl (72 kB)
Collecting wrapt<2,>=1.10
  Using cached wrapt-1.14.1-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (81 kB)
Collecting attrs>=17.4.0
  Using cached attrs-21.4.0-py2.py3-none-any.whl (60 kB)
Collecting importlib-resources>=1.4.0
  Using cached importlib_resources-5.7.1-py3-none-any.whl (28 kB)
Collecting pyrsistent!=0.17.0,!=0.17.1,!=0.17.2,>=0.14.0
  Using cached pyrsistent-0.18.1-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (119 kB)
Collecting idna<4,>=2.5
  Using cached idna-3.3-py3-none-any.whl (61 kB)
Collecting certifi>=2017.4.17
  Using cached certifi-2022.5.18.1-py3-none-any.whl (155 kB)
Collecting charset-normalizer~=2.0.0
  Using cached charset_normalizer-2.0.12-py3-none-any.whl (39 kB)
Collecting urllib3<1.27,>=1.21.1
  Using cached urllib3-1.26.9-py2.py3-none-any.whl (138 kB)
Collecting zipp>=3.1.0
  Using cached zipp-3.8.0-py3-none-any.whl (5.4 kB)
Installing collected packages: pytz, netaddr, itypes, funcy, zipp, wrapt, urllib3, uritemplate, tablib, svgwrite, sqlparse, six, ruamel.yaml.clib, PyYAML, pyrsistent, pyparsing, pycryptodome, psycopg2-binary, prometheus-client, Pillow, MarkupSafe, Markdown, inflection, idna, gunicorn, click, charset-normalizer, certifi, attrs, async-timeout, asgiref, ruamel.yaml, requests, packaging, Jinja2, importlib-resources, django-prometheus, django-pglocks, Django, deprecated, redis, jsonschema, djangorestframework, django-timezone-field, django-taggit, django-tables2, django-js-asset, django-filter, django-debug-toolbar, django-cors-headers, coreschema, swagger-spec-validator, rq, django-mptt, django-cacheops, coreapi, drf-yasg, django-rq
Successfully installed Django-3.2.6 Jinja2-3.0.1 Markdown-3.3.4 MarkupSafe-2.1.1 Pillow-8.3.1 PyYAML-5.4.1 asgiref-3.5.2 async-timeout-4.0.2 attrs-21.4.0 certifi-2022.5.18.1 charset-normalizer-2.0.12 click-8.1.3 coreapi-2.3.3 coreschema-0.0.4 deprecated-1.2.13 django-cacheops-6.0 django-cors-headers-3.8.0 django-debug-toolbar-3.2.2 django-filter-2.4.0 django-js-asset-2.0.0 django-mptt-0.13.1 django-pglocks-1.0.4 django-prometheus-2.1.0 django-rq-2.4.1 django-tables2-2.4.0 django-taggit-1.5.1 django-timezone-field-4.1.2 djangorestframework-3.12.4 drf-yasg-1.20.0 funcy-1.17 gunicorn-20.1.0 idna-3.3 importlib-resources-5.7.1 inflection-0.5.1 itypes-1.2.0 jsonschema-4.5.1 netaddr-0.8.0 packaging-21.3 prometheus-client-0.14.1 psycopg2-binary-2.9.1 pycryptodome-3.10.1 pyparsing-3.0.9 pyrsistent-0.18.1 pytz-2022.1 redis-4.3.1 requests-2.27.1 rq-1.10.1 ruamel.yaml-0.17.21 ruamel.yaml.clib-0.2.6 six-1.16.0 sqlparse-0.4.2 svgwrite-1.4.1 swagger-spec-validator-2.7.4 tablib-3.0.0 uritemplate-4.1.1 urllib3-1.26.9 wrapt-1.14.1 zipp-3.8.0
Installing local dependencies (pip install -r local_requirements.txt)...
Collecting netbox-topology-views
  Using cached netbox_topology_views-1.2.0-py3-none-any.whl (317 kB)
Collecting ntc-netbox-plugin-onboarding
  Using cached ntc_netbox_plugin_onboarding-2.2.0-py3-none-any.whl (599 kB)
Collecting napalm<4,>=2.5.0
  Using cached napalm-3.4.1-py2.py3-none-any.whl (256 kB)
Requirement already satisfied: zipp<4.0.0,>=3.4.0 in ./venv/lib/python3.8/site-packages (from ntc-netbox-plugin-onboarding->-r local_requirements.txt (line 2)) (3.8.0)
Collecting scp
  Using cached scp-0.14.4-py2.py3-none-any.whl (8.6 kB)
Collecting cffi>=1.11.3
  Using cached cffi-1.15.0-cp38-cp38-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (446 kB)
Collecting ncclient
  Using cached ncclient-0.6.13-py2.py3-none-any.whl
Requirement already satisfied: jinja2 in ./venv/lib/python3.8/site-packages (from napalm<4,>=2.5.0->ntc-netbox-plugin-onboarding->-r local_requirements.txt (line 2)) (3.0.1)
Requirement already satisfied: pyYAML in ./venv/lib/python3.8/site-packages (from napalm<4,>=2.5.0->ntc-netbox-plugin-onboarding->-r local_requirements.txt (line 2)) (5.4.1)
Collecting pyeapi>=0.8.2
  Using cached pyeapi-0.8.4-py3-none-any.whl
Collecting netmiko<4.0.0,>=3.3.0
  Using cached netmiko-3.4.0-py3-none-any.whl (178 kB)
Collecting junos-eznc>=2.6.3
  Using cached junos_eznc-2.6.3-py2.py3-none-any.whl (196 kB)
Collecting lxml>=4.3.0
  Using cached lxml-4.8.0-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.manylinux_2_24_x86_64.whl (6.9 MB)
Collecting future
  Using cached future-0.18.2-py3-none-any.whl
Collecting textfsm
  Using cached textfsm-1.1.2-py2.py3-none-any.whl (44 kB)
Requirement already satisfied: requests>=2.7.0 in ./venv/lib/python3.8/site-packages (from napalm<4,>=2.5.0->ntc-netbox-plugin-onboarding->-r local_requirements.txt (line 2)) (2.27.1)
Collecting paramiko>=2.6.0
  Using cached paramiko-2.11.0-py2.py3-none-any.whl (212 kB)
Collecting ciscoconfparse
  Using cached ciscoconfparse-1.6.40-py3-none-any.whl (136 kB)
Requirement already satisfied: netaddr in ./venv/lib/python3.8/site-packages (from napalm<4,>=2.5.0->ntc-netbox-plugin-onboarding->-r local_requirements.txt (line 2)) (0.8.0)
Requirement already satisfied: setuptools>=38.4.0 in ./venv/lib/python3.8/site-packages (from napalm<4,>=2.5.0->ntc-netbox-plugin-onboarding->-r local_requirements.txt (line 2)) (44.0.0)
Collecting pycparser
  Using cached pycparser-2.21-py2.py3-none-any.whl (118 kB)
Collecting yamlordereddictloader
  Using cached yamlordereddictloader-0.4.0-py3-none-any.whl
Collecting ncclient
  Using cached ncclient-0.6.9-py2.py3-none-any.whl
Collecting pyserial
  Using cached pyserial-3.5-py2.py3-none-any.whl (90 kB)
Requirement already satisfied: six in ./venv/lib/python3.8/site-packages (from junos-eznc>=2.6.3->napalm<4,>=2.5.0->ntc-netbox-plugin-onboarding->-r local_requirements.txt (line 2)) (1.16.0)
Requirement already satisfied: pyparsing in ./venv/lib/python3.8/site-packages (from junos-eznc>=2.6.3->napalm<4,>=2.5.0->ntc-netbox-plugin-onboarding->-r local_requirements.txt (line 2)) (3.0.9)
Collecting transitions
  Using cached transitions-0.8.11-py2.py3-none-any.whl (83 kB)
Requirement already satisfied: MarkupSafe>=2.0 in ./venv/lib/python3.8/site-packages (from jinja2->napalm<4,>=2.5.0->ntc-netbox-plugin-onboarding->-r local_requirements.txt (line 2)) (2.1.1)
Collecting tenacity
  Using cached tenacity-8.0.1-py3-none-any.whl (24 kB)
Collecting ntc-templates
  Using cached ntc_templates-3.0.0-py3-none-any.whl (303 kB)
Collecting cryptography>=2.5
  Using cached cryptography-37.0.2-cp36-abi3-manylinux_2_24_x86_64.whl (4.0 MB)
Collecting pynacl>=1.0.1
  Using cached PyNaCl-1.5.0-cp36-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.manylinux_2_24_x86_64.whl (856 kB)
Collecting bcrypt>=3.1.3
  Using cached bcrypt-3.2.2-cp36-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.manylinux_2_24_x86_64.whl (62 kB)
Requirement already satisfied: certifi>=2017.4.17 in ./venv/lib/python3.8/site-packages (from requests>=2.7.0->napalm<4,>=2.5.0->ntc-netbox-plugin-onboarding->-r local_requirements.txt (line 2)) (2022.5.18.1)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in ./venv/lib/python3.8/site-packages (from requests>=2.7.0->napalm<4,>=2.5.0->ntc-netbox-plugin-onboarding->-r local_requirements.txt (line 2)) (1.26.9)
Requirement already satisfied: idna<4,>=2.5 in ./venv/lib/python3.8/site-packages (from requests>=2.7.0->napalm<4,>=2.5.0->ntc-netbox-plugin-onboarding->-r local_requirements.txt (line 2)) (3.3)
Requirement already satisfied: charset-normalizer~=2.0.0 in ./venv/lib/python3.8/site-packages (from requests>=2.7.0->napalm<4,>=2.5.0->ntc-netbox-plugin-onboarding->-r local_requirements.txt (line 2)) (2.0.12)
Collecting loguru==0.6.0
  Using cached loguru-0.6.0-py3-none-any.whl (58 kB)
Collecting dnspython<3.0.0,>=2.1.0
  Using cached dnspython-2.2.1-py3-none-any.whl (269 kB)
Collecting passlib<2.0.0,>=1.7.4
  Using cached passlib-1.7.4-py2.py3-none-any.whl (525 kB)
Collecting toml==0.10.2
  Using cached toml-0.10.2-py2.py3-none-any.whl (16 kB)
Installing collected packages: pyserial, passlib, netbox-topology-views, yamlordereddictloader, transitions, toml, tenacity, pyeapi, pycparser, lxml, loguru, future, dnspython, textfsm, ciscoconfparse, cffi, pynacl, ntc-templates, cryptography, bcrypt, paramiko, scp, ncclient, netmiko, junos-eznc, napalm, ntc-netbox-plugin-onboarding
Successfully installed bcrypt-3.2.2 cffi-1.15.0 ciscoconfparse-1.6.40 cryptography-37.0.2 dnspython-2.2.1 future-0.18.2 junos-eznc-2.6.3 loguru-0.6.0 lxml-4.8.0 napalm-3.4.1 ncclient-0.6.9 netbox-topology-views-1.2.0 netmiko-3.4.0 ntc-netbox-plugin-onboarding-2.2.0 ntc-templates-3.0.0 paramiko-2.11.0 passlib-1.7.4 pycparser-2.21 pyeapi-0.8.4 pynacl-1.5.0 pyserial-3.5 scp-0.14.4 tenacity-8.0.1 textfsm-1.1.2 toml-0.10.2 transitions-0.8.11 yamlordereddictloader-0.4.0
Applying database migrations (python3 netbox/manage.py migrate)...
Traceback (most recent call last):
  File "netbox/manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/ceph-g/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/ceph-g/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/django/core/management/__init__.py", line 413, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/ceph-g/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/django/core/management/base.py", line 354, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/ceph-g/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/django/core/management/base.py", line 398, in execute
    output = self.handle(*args, **options)
  File "/ceph-g/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/django/core/management/base.py", line 89, in wrapped
    res = handle_func(*args, **kwargs)
  File "/ceph-g/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/django/core/management/commands/migrate.py", line 75, in handle
    self.check(databases=[database])
  File "/ceph-g/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/django/core/management/base.py", line 419, in check
    all_issues = checks.run_checks(
  File "/ceph-g/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/django/core/checks/registry.py", line 76, in run_checks
    new_errors = check(app_configs=app_configs, databases=databases)
  File "/ceph-g/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/django/core/checks/urls.py", line 13, in check_url_config
    return check_resolver(resolver)
  File "/ceph-g/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/django/core/checks/urls.py", line 23, in check_resolver
    return check_method()
  File "/ceph-g/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/django/urls/resolvers.py", line 412, in check
    for pattern in self.url_patterns:
  File "/ceph-g/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/ceph-g/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/django/urls/resolvers.py", line 598, in url_patterns
    patterns = getattr(self.urlconf_module, "urlpatterns", self.urlconf_module)
  File "/ceph-g/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/ceph-g/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/django/urls/resolvers.py", line 591, in urlconf_module
    return import_module(self.urlconf_name)
  File "/usr/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 848, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/ceph-g/opt/netbox-2.11.12/netbox/netbox/urls.py", line 8, in <module>
    from extras.plugins.urls import plugin_admin_patterns, plugin_patterns, plugin_api_patterns
  File "/ceph-g/opt/netbox-2.11.12/netbox/extras/plugins/urls.py", line 28, in <module>
    urlpatterns = import_object(f"{plugin_path}.urls.urlpatterns")
  File "/ceph-g/opt/netbox-2.11.12/netbox/extras/plugins/utils.py", line 31, in import_object
    spec.loader.exec_module(module)
  File "/ceph-g/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/netbox_topology_views/urls.py", line 3, in <module>
    from . import views
  File "/ceph-g/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/netbox_topology_views/views.py", line 9, in <module>
    from .forms import DeviceFilterForm
  File "/ceph-g/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/netbox_topology_views/forms.py", line 13, in <module>
    from netbox.forms import NetBoxModelFilterSetForm
**ImportError: cannot import name 'NetBoxModelFilterSetForm' from 'netbox.forms' (/opt/netbox-2.11.12/netbox/netbox/forms.py)**

**Comment by kkthxbye-code** [comment, marked as answer]
The version of `netbox_topology_views` installed doesn't support the version of netbox you are trying to update to. Either disable it while upgrading or try to match the version with one that works with 2.11.12.
